pkg/operator/ceph/object/shared_pools.go

```diff
@@ -0,0 +1,695 @@
+		return false
+	}
+	return true
+}
+
+func validatePoolPlacements(placements []cephv1.PoolPlacementSpec) error {
+	names := make(map[string]struct{}, len(placements))
+	for _, p := range placements {
+		if _, ok := names[p.Name]; ok {
+			return fmt.Errorf("invalidObjStorePoolCofig: invalid placement %s: placement names must be unique", p.Name)
+		}
+		names[p.Name] = struct{}{}
+		if err := validatePoolPlacementStorageClasses(p.StorageClasses); err != nil {
+			return fmt.Errorf("invalidObjStorePoolCofig: invalid placement %s: %w", p.Name, err)
+		}
+	}
+	return nil
+}
+
+func validatePoolPlacementStorageClasses(scList []cephv1.PlacementStorageClassSpec) error {
+	names := make(map[string]struct{}, len(scList))
+	for _, sc := range scList {
+		if sc.Name == defaultPlacementStorageClass {
+			return fmt.Errorf("invalid placement StorageClass %s: %s name is reserved", sc.Name, defaultPlacementStorageClass)
+		}
+		if _, ok := names[sc.Name]; ok {
+			return fmt.Errorf("invalid placement StorageClass %s: name must be unique", sc.Name)
+		}
+		names[sc.Name] = struct{}{}
+	}
+	return nil
+}
+
+func sharedPoolsExist(objContext *Context, sharedPools cephv1.ObjectSharedPoolsSpec) error {
+	existingPools, err := cephclient.ListPoolSummaries(objContext.Context, objContext.clusterInfo)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list pools")
+	}
+	existing := make(map[string]struct{}, len(existingPools))
+	for _, pool := range existingPools {
+		existing[pool.Name] = struct{}{}
+	}
+	// ignore pools with empty name:
+	existing[""] = struct{}{}
+
+	if _, ok := existing[sharedPools.MetadataPoolName]; !ok {
+		return fmt.Errorf("sharedPool do not exist: %s", sharedPools.MetadataPoolName)
+	}
+	if _, ok := existing[sharedPools.DataPoolName]; !ok {
+		return fmt.Errorf("sharedPool do not exist: %s", sharedPools.DataPoolName)
+	}
+
+	for _, pp := range sharedPools.PoolPlacements {
+		if _, ok := existing[pp.MetadataPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.MetadataPoolName, pp.Name)
+		}
+		if _, ok := existing[pp.DataPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.DataPoolName, pp.Name)
+		}
+		if _, ok := existing[pp.DataNonECPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.DataNonECPoolName, pp.Name)
+		}
+		for _, sc := range pp.StorageClasses {
+			if _, ok := existing[sc.DataPoolName]; !ok {
+				return fmt.Errorf("sharedPool do not exist: pool %s for StorageClass %s", sc.DataPoolName, sc.Name)
+			}
+		}
+	}
+
+	logger.Info("verified shared pools exist")
+	return nil
+}
+
+func ConfigureSharedPoolsForZone(objContext *Context, sharedPools cephv1.ObjectSharedPoolsSpec) error {
+	if sharedPools.DataPoolName == "" && sharedPools.MetadataPoolName == "" && len(sharedPools.PoolPlacements) == 0 {
+		logger.Debugf("no shared pools to configure for store %q", objContext.Name)
+		return nil
+	}
+
```

Comment:
Instead of some of the line-by-line debugging here (which can get a little too verbose to debug), I would suggest a high level info line here

```
logger.Infof("configuring shared pools for object store %q", namespacedName)
```